### PR TITLE
VIDLA-4391 - Support for Multiple Players with Embedded Prebid Plugin on a Page

### PR DESCRIPTION
**Pull request by graphicdeveloper**
This merge adds support for multiple Brightcove players with an embedded Prebid Plugin on the same html page (when players are NOT using iframe embeds).

### CODE REVIEW

File: src/BcPrebidLoader.js [hunk 1]
```diff
@@ -3,171 +3,223 @@
...
 
-// register videojs prebid plugins
-function regPrebidVastPlugin(vjs) {
-	_vjs = vjs;
-	// getPlugins not exist in Brightcove Player v5.28.1
-	if (!vjs.getPlugins || !_vjs.getPlugins().bcPrebidVastPlugin) {
-		registerPrebidVastPlugin();
-	}
+// STATE
+window.BCVideo_PrebidPluginApiQue = null;
```
Comment (ynisman) [inline src/BcPrebidLoader.js]: Can we use $$PREBID_GLOBAL$$.BCVideo_PrebidPluginApiQue ?

File: src/BcPrebidLoader.js [hunk 2]
```diff
@@ -3,171 +3,223 @@
...
+		if ($$PREBID_GLOBAL$$.plugin_prebid_options) {
+			var prebidPluginPath = $$PREBID_GLOBAL$$.plugin_prebid_options.prebidPluginPath ? $$PREBID_GLOBAL$$.plugin_prebid_options.prebidPluginPath : DEFAULT_PLUGIN_JS_URL;
 
-	doPrebid: function(options, id) {
-		this.id = id;
-		options.onlyPrebid = true;
-		_vjs(id).bcPrebidVastPlugin(options);
-	},
+			var runPlugin = function () {
+				// var prebidPluginObj = new BCVideo_PrebidVastMainPlugin(playerRef);
```
[the remark this answers is not in the document]
Comment (graphicdeveloper) [reply src/BcPrebidLoader.js]: Good catch, fixed.

File: src/BcPrebidLoader.js (hunk 1 above)
Comment (graphicdeveloper) [reply src/BcPrebidLoader.js]: Sure, I think that will work. :-)

File: src/BcPrebidVast.js [hunk 3]
```diff
@@ -583,10 +651,16 @@ var prebidVastPlugin = function(player) {
...
+                _adListManagerObj.stop();
+            }
 		}
 	};
 };
 // ////////////////////////////////////////////////////////////////////
 // EXPORTS
 module.exports = prebidVastPlugin;
+
+if (window.BCVideo_PrebidPluginApiQue) {
```
[the remark this answers is not in the document]
Comment (graphicdeveloper) [reply src/BcPrebidVast.js]: FIxed.

File: src/BcPrebidLoader.js [hunk 4]
```diff
@@ -3,171 +3,223 @@
...
+        _options = options;
+
+        var runPlugin = function () {
+            var apiFunc = getLoadedPluginAPI();
+            _prebidPluginObj = apiFunc(player);					// uses local var player
+            _prebidPluginObj.run(options);						// uses local var options
+        };
+
+        var handleError = function () {
+            // TODO: Error handling for main plugin loading error
```
[the remark this answers is not in the document]
Comment (graphicdeveloper) [reply src/BcPrebidLoader.js]: Placeholder for more robust error handling if we want to add it later

**Pull request merged**